### PR TITLE
Removed unnecessary newTestDatabase function

### PR DESCRIPTION
**Pull request by zhelyabuzhsky**
There were two places where you needed to change the connection string to the test database. It was not very elegant.

### CODE REVIEW

File: database/redis/selfstate_test.go [hunk 1]
```diff
@@ -65,7 +66,7 @@ func TestSelfCheckWithWritesNotInChecker(t *testing.T) {
 
 func testSelfCheckWithWritesInDBSource(t *testing.T, dbSource DBSource) {
 	logger, _ := logging.GetLogger("dataBase")
-	dataBase := NewDatabase(logger, config, dbSource)
+	dataBase := NewTestDatabase(logger)
```
Comment (dmitryanchikov) [inline database/redis/selfstate_test.go:69]: Probably you'd want to add like you add above for `Checker`:
```
	dataBase.source = dbSource 
```

File: database/redis/database.go [hunk 2]
```diff
@@ -79,7 +79,14 @@ func NewDatabase(logger moira.Logger, config Config, source DBSource) *DbConnect
 
 // NewTestDatabase use it only for tests
 func NewTestDatabase(logger moira.Logger) *DbConnector {
-	return NewDatabase(logger, Config{Addrs: []string{"0.0.0.0:6379"}}, testSource)
+	return NewDatabase(logger, Config{
+		Addrs: []string{"0.0.0.0:6379"},
+	}, testSource)
+}
```
Comment (dmitryanchikov) [inline database/redis/database.go:85]: I see that some tests rely on existing dbSource in test daatbase object, would it make more sense to add it as argument to `NewTestDatabase` func with `default=testSource`?

File: database/redis/database_test.go [hunk 3]
```diff
@@ -4,25 +4,16 @@ import (
...
-
-func newTestDatabase(logger moira.Logger, config Config) *DbConnector {
-	return NewDatabase(logger, config, testSource)
-}
-
 func TestNewDatabase(t *testing.T) {
 	Convey("NewDatabase should return correct DBConnector", t, func() {
 		logger, _ := logging.ConfigureLog("stdout", "info", "test", true) // nolint: govet
-		database := NewDatabase(logger, config, testSource)
+		database := NewTestDatabase(logger)
```
Comment (dmitryanchikov) [inline database/redis/database_test.go:16]: `dbSource=testSource` as well?

**Review by dmitryanchikov — COMMENTED** [review]
Looks great, just a few questions

File: database/redis/database_test.go (hunk 3 above)
Comment (dmitryanchikov) [reply database/redis/database_test.go:16]: Disregard, I see tht internally `NewTestDatabase` already set `testSource`, so it is good

File: database/redis/database.go (hunk 2 above)
Comment (zhelyabuzhsky) [reply database/redis/database.go:85]: There are not default arguments in Go )

File: database/redis/selfstate_test.go (hunk 1 above)
Comment (zhelyabuzhsky) [reply database/redis/selfstate_test.go:69]: fixed

**Review by dmitryanchikov — APPROVED** [review]

**Review by KiskachiMaria — APPROVED** [review]

**Pull request merged**